forms/ui/ui_publishdialog.h

```diff
@@ -160,28 +177,54 @@ class Ui_PublishDialog
         label_5->setFont(font1);
         label_5->setTextFormat(Qt::AutoText);
 
-        gridLayout_5->addWidget(label_5, 1, 0, 1, 1);
+        verticalLayout->addWidget(label_5);
 
         frame = new QFrame(PublishDialog);
         frame->setObjectName(QStringLiteral("frame"));
         frame->setFrameShape(QFrame::StyledPanel);
         frame->setFrameShadow(QFrame::Raised);
-        gridLayout_4 = new QGridLayout(frame);
-        gridLayout_4->setObjectName(QStringLiteral("gridLayout_4"));
+        outer_girdLayout = new QGridLayout(frame);
+        outer_girdLayout->setObjectName(QStringLiteral("outer_girdLayout"));
         spin_HistoryQos = new QSpinBox(frame);
         spin_HistoryQos->setObjectName(QStringLiteral("spin_HistoryQos"));
         spin_HistoryQos->setMinimum(1);
         spin_HistoryQos->setMaximum(200);
         spin_HistoryQos->setValue(1);
 
-        gridLayout_4->addWidget(spin_HistoryQos, 0, 1, 1, 1);
+        outer_girdLayout->addWidget(spin_HistoryQos, 0, 1, 1, 1);
+
+        label_6 = new QLabel(frame);
+        label_6->setObjectName(QStringLiteral("label_6"));
+        label_6->setFont(font);
+        label_6->setTextFormat(Qt::AutoText);
+
+        outer_girdLayout->addWidget(label_6, 0, 0, 1, 1);
+
+        checkBox_reliable = new QCheckBox(frame);
+        checkBox_reliable->setObjectName(QStringLiteral("checkBox_reliable"));
+        checkBox_reliable->setFont(font);
+        checkBox_reliable->setChecked(true);
+
+        outer_girdLayout->addWidget(checkBox_reliable, 0, 2, 1, 1);
+
+        label_10 = new QLabel(frame);
+        label_10->setObjectName(QStringLiteral("label_10"));
+        label_10->setFont(font);
+        label_10->setTextFormat(Qt::AutoText);
+
+        outer_girdLayout->addWidget(label_10, 1, 0, 1, 1);
+
+        comboBox_durability = new QComboBox(frame);
+        comboBox_durability->setObjectName(QStringLiteral("comboBox_durability"));
+
+        outer_girdLayout->addWidget(comboBox_durability, 1, 1, 1, 1);
```

Comment:
The widgets above, are they not currently in the GUI? Or has the code been moved here from somewhere else?

Reply:
I just order the widgets in the file according to the GUI layout